scripts/extract_base_urls.py

```
@@ -0,0 +1,199 @@
+    """
+    try:
+        parsed = urlparse(full_url)
+        
+        # Start with the domain
+        base = parsed.netloc
+        
+        # Add the first path segment if it exists
+        if parsed.path and parsed.path != "/":
+            path_parts = [part for part in parsed.path.split("/") if part]
+            if path_parts:
+                # For common cases, include more meaningful path segments
```

Ruff (0.12.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)